### PR TITLE
getipaddrs: make loopback (Bool) a keyword argument

### PR DESCRIPTION
**Pull request by StefanKarpinski**
I think this is a somewhat better API. Boolean arguments are often better as keywords since the value itself conveys so little contextual information. The loopback argument and the type argument are also orthogonal which this allows us to express slightly more easily (i.e. with a single method definition).

### CODE REVIEW

File: stdlib/Sockets/src/addrinfo.jl [hunk 1]
```diff
@@ -267,7 +269,7 @@ julia> getipaddrs(IPv6)
  ip"fe80::445e:5fff:fe5d:5500"
 ```
 """
-function getipaddrs(addr_type::Type{T}, include_lo::Bool=false) where T<:IPAddr
+function getipaddrs(addr_type::Type{T}=IPAddr; loopback::Bool=false) where T<:IPAddr
```
Comment (jpsamaroo) [inline stdlib/Sockets/src/addrinfo.jl]: Thanks for adding in the 0-argument method, seems like I forgot that in my PR. :heart:

Comment (StefanKarpinski) [reply stdlib/Sockets/src/addrinfo.jl]: You had it, it just couldn’t be specified like this because of the interaction with non-keyword loopback argument.

**Comment by StefanKarpinski** [comment]
AppVeyor and Travis failures don't seem related but man does the CI situation not make me super comfortable to merge anything.

**Comment by StefanKarpinski** [comment]
I want to make sure this gets into the 1.2 release since this is a change to a new API and if it doesn't make it into 1.2 we can't change this in 1.3.

**Comment by jpsamaroo** [comment]
Unrelated, but `s/gitipaddrs/getipaddrs/` in your commit message

**Comment by StefanKarpinski** [comment]
Reminder to self: squash this when you merge it and fix the commit message.

**Pull request merged**